lib/spack/spack/environment/environment.py

```diff
@@ -2230,11 +2213,10 @@ def write(self, regenerate: bool = True) -> None:
             self.regenerate_views()
             spack.hooks.post_env_write(self)
 
-        self._reset_new_specs_and_installs()
+        self._reset_new_specs()
 
-    def _reset_new_specs_and_installs(self) -> None:
+    def _reset_new_specs(self) -> None:
```

Comment:
Should this really be a method at this point? If so, then we should also be calling it at line 931 in the `_read` method.